tests/client_server.rs

```diff
@@ -103,6 +103,12 @@ fn client_server_integration() {
 
 fn standard_client_tests(valgrind: Option<String>) -> Vec<ClientTest> {
     vec![
+        ClientTest {
+            name: "rustls-platform-verifier",
+            valgrind: valgrind.clone(),
+            env: vec![("RUSTLS_PLATFORM_VERIFIER", "1")],
+            expect_error: true,
```

Comment:
it might be nice to think of a way we could test this against a server with a system trusted certificate. Probably the easiest way is an online test (for e.g. against `example.com`). Maybe a separate `Makefile` PHONY target for this? Something like `make connect-tests` ?

Reply:
Good idea, will do :+1:

Reply:
Added a new test to `client_server.rs` by slightly enriching the existing `ClientTest`, but let me know if this is undesirable, happy to also add a new Makefile target :+1:

Reply:
I think a separate Makefile target would be preferable if you don't mind reworking it. 

It feels separate from the traditional integration tests because we're testing just the client against unknown server software (vs our own server binary). I also think it could flake on occasion from bad internet weather and other factors out of our control. Being able to quickly disable just the `example.com` part could be helpful.

Reply:
Should it be run on CI?

Reply:
My vote would be yes, but as a separate job after the ones running the Rust-based integration tests like:

```
- name: Platform verifier connect test
  run: make connect-test
```

Where the `connect-test` PHONY target will depend on the `client` target, and run the client binary with the right env flags/args to connect to `example.com` with the verifier feature enabled. We'll consider the exit code being a success as sufficient test coverage.

Does that make sense? It's possible I'm overlooking a detail that could make this less appealing than I think.

Reply:
Added a Makefile target which is run in CI. It currently isn't being run in Windows CI (as that uses cmake), maybe that's fine.

Reply:
Sounds OK to me. I haven't done my homework on `CMake` to learn what the equivalent setup would look like.